c-layer/lib.cxx

```diff
@@ -422,4 +425,17 @@ void build_repository_cache(const char *alias, struct Status *status, ZyppProgre
     status->error = strdup(excpt.asUserString().c_str());
   }
 }
+
+void import_gpg_key(const char* const pathname, struct Status *status) noexcept {
+  try {
+    zypp::filesystem::Pathname path(pathname);
+    zypp::PublicKey key(path);
+    zypp_ptr()->keyRing()->importKey(key, true); // by default trust keys unless we need to distinguish it
```

Comment:
This smells like a security problem. We need a better comment, if not better code.

Related YaST code?
https://github.com/yast/yast-pkg-bindings/blob/f2e89d76a3a97da0244421316758c358dde68ce9/src/Keyring.cc#L65
https://github.com/search?q=org%3Ayast+ImportGPGKey&type=code

Reply:
yeap. but related agama code is only https://github.com/agama-project/agama/blob/master/service/lib/agama/software/manager.rb#L459
so in general we use it only to import to our separated zypp target what was already trusted in root. If we need to distinguish it, we can always add that parameter

Reply:
This is basically importing the key in advance. You can import a GPG key into the libzypp keyring *before* adding a repository. Then libzypp already knows the key and does not ask via callbacks.

If you do not do that in advance libzypp will ask to import the key using a callback when it finds an unknown key during repository refresh.

A GPG key can be known by libzypp (i.e. saved in the libzypp data) but not trusted. Then libzypp will reject all data signed by this key. If a key is trusted then libzypp will use for verifying the signatures.

Reply:
Thank you, code comment added

Reply:
I would say that probably now for untrusted version we do not have usage...but maybe in future for unattended installation to "revoke" known leaked keys.

Reply:
... and deleted, according to [KeyRing tests](https://github.com/openSUSE/libzypp/blob/747ae97eebddfe3404e75047edb70fc71d234576/tests/zypp/KeyRing_test.cc#L103) importing an untrusted key means something else... and I don't really care what.
Will make another code comment